brpc-java-core/src/main/java/com/baidu/brpc/spi/ExtensionLoaderManager.java

```diff
@@ -24,16 +23,21 @@ public static ExtensionLoaderManager getInstance() {
         return instance;
     }
 
-    private AtomicBoolean load = new AtomicBoolean(false);
+    private Object loadLock = new Object();
+    private Boolean isLoaded = false;
 
     /**
      * load all extensions with java spi
      */
     public void loadAllExtensions(String encoding) {
-        if (load.compareAndSet(false, true)) {
-            loadNamingService();
-            loadProtocol(encoding);
-            loadLoadBalance();
+        synchronized (loadLock){
```

Comment:
it's better to double check if isLoaded is true as following:
if (!isLoaded) {
  synchronized (loadLock) {
       if (!isLoaded) {
       }
  }
}